build fails with master at e5833c182cd05b15fe3450bc792abfe34e7e52dd

### Summary

Build error with master at e5833c182cd05b15fe3450bc792abfe34e7e52dd



### System information

Mbed TLS version (number or commit id): e5833c182cd05b15fe3450bc792abfe34e7e52dd
Operating system and version:  Fedora 36
Configuration (if not default, please attach `mbedtls_config.h`): 
Compiler and options (if you used a pre-built binary, please indicate how you obtained it): 
Additional environment information: 

### Expected behavior

The build shall be clean.  Looks like the test seem intended but one could set garbage data as initializer to silence the error.

### Actual behavior



### Steps to reproduce

1. clone new repo.
2. mkdir build.
3. cd build.
4. cmake  ..
5. make




### Additional information

Below is the compiler error:

```bash
[ 65%] Building C object tests/CMakeFiles/test_suite_ecdh.dir/test_suite_ecdh.c.o
/home/devnaga/work/mbedtls/tests/suites/test_suite_ecdh.function: In function ‘test_ecdh_invalid_param’:
/home/devnaga/work/mbedtls/tests/suites/test_suite_ecdh.function:54:29: error: ‘kp’ may be used uninitialized [-Werror=maybe-uninitialized]
   54 |                             mbedtls_ecdh_get_params( &ctx, &kp,
      |                             ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/devnaga/work/mbedtls/tests/suites/test_suite_ecdh.function:2:
/home/devnaga/work/mbedtls/include/mbedtls/ecdh.h:347:5: note: by argument 2 of type ‘const mbedtls_ecp_keypair *’ to ‘mbedtls_ecdh_get_params’ declared here
  347 | int mbedtls_ecdh_get_params( mbedtls_ecdh_context *ctx,
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/home/devnaga/work/mbedtls/tests/suites/test_suite_ecdh.function:50:25: note: ‘kp’ declared here
   50 |     mbedtls_ecp_keypair kp;
      |                         ^~
cc1: all warnings being treated as errors
make[2]: *** [tests/CMakeFiles/test_suite_ecdh.dir/build.make:86: tests/CMakeFiles/test_suite_ecdh.dir/test_suite_ecdh.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3798: tests/CMakeFiles/test_suite_ecdh.dir/all] Error 2

```

--- a/tests/suites/test_suite_ecdh.function
+++ b/tests/suites/test_suite_ecdh.function
@@ -65,6 +65,8 @@ void ecdh_invalid_param( )
     int invalid_side = 42;
     mbedtls_ecp_group_id valid_grp = MBEDTLS_ECP_DP_SECP192R1;
 
+    mbedtls_ecp_keypair_init( &kp );
+    mbedtls_ecdh_init( &ctx );
     TEST_INVALID_PARAM( mbedtls_ecdh_init( NULL ) );
 
 #if defined(MBEDTLS_ECP_RESTARTABLE)

--- a/tests/suites/test_suite_ecdsa.function
+++ b/tests/suites/test_suite_ecdsa.function
@@ -20,6 +20,12 @@ void ecdsa_invalid_param( )
     size_t slen;
     unsigned char buf[42] = { 0 };
 
+    mbedtls_ecdsa_init( &ctx );
+    mbedtls_ecp_keypair_init( &key );
+    mbedtls_ecp_group_init( &grp );
+    mbedtls_ecp_point_init( &P );
+    mbedtls_mpi_init( &m );
+
     TEST_INVALID_PARAM( mbedtls_ecdsa_init( NULL ) );
     TEST_VALID_PARAM( mbedtls_ecdsa_free( NULL ) );
 
@@ -200,6 +206,12 @@ void ecdsa_invalid_param( )
                        mbedtls_ecdsa_from_keypair( &ctx, NULL ) );
 
 exit:
+    mbedtls_ecdsa_free( &ctx );
+    mbedtls_ecp_keypair_free( &key );
+    mbedtls_ecp_group_free( &grp );
+    mbedtls_ecp_point_free( &P );
+    mbedtls_mpi_free( &m );
+
     return;
 }
 /* END_CASE */

--- a/tests/suites/test_suite_ecjpake.function
+++ b/tests/suites/test_suite_ecjpake.function
@@ -110,6 +110,8 @@ void ecjpake_invalid_param( )
     mbedtls_md_type_t valid_md = MBEDTLS_MD_SHA256;
     mbedtls_ecp_group_id valid_group = MBEDTLS_ECP_DP_SECP256R1;
 
+    mbedtls_ecjpake_init( &ctx );
+
     TEST_INVALID_PARAM( mbedtls_ecjpake_init( NULL ) );
     TEST_VALID_PARAM( mbedtls_ecjpake_free( NULL ) );
 

--- a/tests/suites/test_suite_ecp.function
+++ b/tests/suites/test_suite_ecp.function
@@ -57,6 +57,9 @@ void ecp_invalid_param( )
     mbedtls_ecp_restart_ctx restart_ctx;
 #endif /* MBEDTLS_ECP_RESTARTABLE */
 
+    mbedtls_ecp_group_init( &grp );
+    mbedtls_ecp_point_init( &P );
+
     TEST_INVALID_PARAM( mbedtls_ecp_point_init( NULL ) );
     TEST_INVALID_PARAM( mbedtls_ecp_keypair_init( NULL ) );
     TEST_INVALID_PARAM( mbedtls_ecp_group_init( NULL ) );

